streams/src/test/java/org/apache/kafka/streams/state/internals/metrics/StateStoreMetricsTest.java

```diff
@@ -327,6 +327,38 @@ public void shouldGetExpiredWindowRecordDropSensor() {
         assertThat(sensor, is(expectedSensor));
     }
 
+    @Test
+    public void shouldGetRecordE2ELatencySensor() {
+        final String metricName = "record-e2e-latency";
+
+        final String e2eLatencyDescription =
+            "end-to-end latency of a record, measuring by comparing the record timestamp with the "
+                + "system time when it has been fully processed by the node";
+        final String descriptionOfAvg = "The average " + e2eLatencyDescription;
+        final String descriptionOfMin = "The minimum " + e2eLatencyDescription;
+        final String descriptionOfMax = "The maximum " + e2eLatencyDescription;
+
+        expect(streamsMetrics.storeLevelSensor(THREAD_ID, TASK_ID, STORE_NAME, metricName, RecordingLevel.TRACE))
+            .andReturn(expectedSensor);
+        expect(streamsMetrics.storeLevelTagMap(THREAD_ID, TASK_ID, STORE_TYPE, STORE_NAME)).andReturn(storeTagMap);
+        StreamsMetricsImpl.addAvgAndMinAndMaxToSensor(
+            expectedSensor,
+            STORE_LEVEL_GROUP,
```

Comment:
You need to make this dependent on the built-in metrics version by using instance variable `storeLevelGroup`.

Reply:
I just realized that we should not put new metrics into old groups. Your code is fine. Do not use instance variable `storeLevelGroup`! Sorry for the confusion.